### PR TITLE
fix to avoid circular imports when using octree

### LINKED ISSUE #2834: napari will not run with octree selected
## 🐛 Bug

I cannot start up napari with NAPARI_OCTREE=1.  There is a circular import happening...

```
Traceback (most recent call last):
  File "/Users/pwadhwa/anaconda3/envs/napari-env/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/pwadhwa/anaconda3/envs/napari-env/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/pwadhwa/Documents/repos/napari/napari/__main__.py", line 418, in <module>
    sys.exit(main())
  File "/Users/pwadhwa/Documents/repos/napari/napari/__main__.py", line 414, in main
    _run()
  File "/Users/pwadhwa/Documents/repos/napari/napari/__main__.py", line 225, in _run
    from napari import run, view_path
  File "/Users/pwadhwa/Documents/repos/napari/napari/_lazy.py", line 46, in __getattr__
    from .utils import _magicgui
  File "/Users/pwadhwa/Documents/repos/napari/napari/utils/_magicgui.py", line 16, in <module>
    from .. import layers, types
  File "/Users/pwadhwa/Documents/repos/napari/napari/layers/__init__.py", line 11, in <module>
    from .image import Image
  File "/Users/pwadhwa/Documents/repos/napari/napari/layers/image/__init__.py", line 1, in <module>
    from .image import Image
  File "/Users/pwadhwa/Documents/repos/napari/napari/layers/image/image.py", line 23, in <module>
    from .experimental._chunked_slice_data import ChunkedSliceData
  File "/Users/pwadhwa/Documents/repos/napari/napari/layers/image/experimental/__init__.py", line 3, in <module>
    from .octree_chunk import OctreeChunk, OctreeChunkGeom, OctreeLocation
  File "/Users/pwadhwa/Documents/repos/napari/napari/layers/image/experimental/octree_chunk.py", line 8, in <module>
    from ....components.experimental.chunk import ChunkLocation, LayerRef
  File "/Users/pwadhwa/Documents/repos/napari/napari/components/__init__.py", line 24, in <module>
    from . import _viewer_key_bindings  # isort:skip
  File "/Users/pwadhwa/Documents/repos/napari/napari/components/_viewer_key_bindings.py", line 5, in <module>
    from .viewer_model import ViewerModel
  File "/Users/pwadhwa/Documents/repos/napari/napari/components/viewer_model.py", line 25, in <module>
    from ..layers import Image, Layer
ImportError: cannot import name 'Image' from partially initialized module 'napari.layers'
 (most likely due to a circular import) (/Users/pwadhwa/Documents/repos/napari/napari/layers/__init__.py)
```

## To Reproduce

Type following---
1.  export NAPARI_OCTREE=1
2.  napari

### PR DESCRIPTION
**Pull request by ppwadhwa**
Fixes #2834 

We will need a test for octree going forward, because its not included currently.

### CODE REVIEW

**Comment by sofroniewn** [comment]
Nice catch!

> We will need a test for octree going forward, because its not included currently.

Agreed, we could either add a new test like `PR Test / ubuntu-18.04 3.8 async-pyqt-py38-linux` but for `octree`. That'll probably be the easiest/ fastest thing to do.

Better would be if we had something a little more like `PR Test / ubuntu-18.04 3.8 experimental-pyqt-py38-linux` that did both async and octree

Even better would just be if async and octree tests happened during our normal tests. We could aim for this but it might be a little trickier.

Thoughts @tlambert03?

File: napari/_lazy.py [hunk 1]
```diff
@@ -42,6 +42,11 @@ def __getattr__(name):
         # see: https://github.com/napari/napari/issues/1347
         from scipy import stats  # noqa: F401
 
+        # Import added to fix circular imports
+        from .viewer import Viewer
```
Comment (tlambert03) [inline napari/_lazy.py:46]: 🤔

very strange!  do you know _why_ this fixes it?

**Comment by tlambert03** [comment]
> Even better would just be if async and octree tests happened during our normal tests. We could aim for this but it might be a little trickier.
> Thoughts @tlambert03?

I don't have any particular thoughts here.  Obviously I'd love if we didn't have to special case so much of the async stuff.  It never did integrate itself into the rest of the program, it doesn't really have any unit tests to speak of, and our "institutional knowledge" is a bit thin.  I don't know what to do with it.

File: napari/_lazy.py (hunk 1 above)
Comment (ppwadhwa) [reply napari/_lazy.py:46]: @sofroniewn figured this one out.  

All I can say about it is that before #2816 was merged, the viewer was imported just before this line was run in the napari `__init`` file:
```
_magicgui.register_types_with_magicgui()
```
so we added it back. 

That's all I got though...

Comment (tlambert03) [reply napari/_lazy.py:46]: ok, lemme see if we can fix this a better way (by moving the magicgui initialization)

**Comment by sofroniewn** [comment]
I think we should go with the approach in #2838 which ultimately cleans a lot of this stuff up more. Thanks for finding though @ppwadhwa!

**Pull request closed without merging**